scm/src/run_gmtb_scm.py

```diff
@@ -83,15 +83,15 @@ def execute(cmd):
     status = p.returncode
     if status == 0:
         message = 'Execution of "{0}" returned with exit code {1}\n'.format(cmd, status)
-        message += '    stdout: "{0}"\n'.format(stdout.rstrip('\n'))
-        message += '    stderr: "{0}"'.format(stderr.rstrip('\n'))
+        message += '    stdout: "{0}"\n'.format(stdout.rstrip('\n'.encode()))
```

Comment:
I thought you need to use `decode`, but maybe I am wrong.

I ended up using the following code (see ccpp-framework's `common.py` script):
```
def execute(cmd, abort = True):
    """Runs a local command in a shell. Waits for completion and
    returns status, stdout and stderr. If abort = True, abort in
    case an error occurs during the execution of the command."""

    # Set debug to true if logging level is debug
    debug = logging.getLogger().getEffectiveLevel() == logging.DEBUG

    logging.debug('Executing "{0}"'.format(cmd))
    p = subprocess.Popen(cmd, stdout = subprocess.PIPE,
                         stderr = subprocess.PIPE, shell = True)
    (stdout, stderr) = p.communicate()
    status = p.returncode
    if debug:
        message = 'Execution of "{0}" returned with exit code {1}\n'.format(cmd, status)
        message += '    stdout: "{0}"\n'.format(stdout.decode(encoding='ascii', errors='ignore').rstrip('\n'))
        message += '    stderr: "{0}"'.format(stderr.decode(encoding='ascii', errors='ignore').rstrip('\n'))
        logging.debug(message)
    if not status == 0:
        message = 'Execution of command {0} failed, exit code {1}\n'.format(cmd, status)
        message += '    stdout: "{0}"\n'.format(stdout.decode(encoding='ascii', errors='ignore').rstrip('\n'))
        message += '    stderr: "{0}"'.format(stderr.decode(encoding='ascii', errors='ignore').rstrip('\n'))
        if abort:
            raise Exception(message)
        else:
            logging.error(message)
    return (status, stdout.decode(encoding='ascii', errors='ignore').rstrip('\n'),
                    stderr.decode(encoding='ascii', errors='ignore').rstrip('\n'))
```

Reply:
Again, this was copied from the master branch that you originally did. This seemed to work on every machine with python2/3 that I tried.

Reply:
Correct, but I had to make changes afterwards because it failed on stampede (and I think also one of the wcoss platforms). Can you test on stampede?

Reply:
Also, the error would only show up if that particular piece of code is executed (since Python is completely ignoring code that doesn't get executed).

Reply:
No, I don't have access to stampede. I can proactively change to this and retest though.

Reply:
I can check that quickly for you, hang on.

Reply:
I could not reproduce the error we saw with the UFS on stampede, so I suggest to leave it as it is.

Reply:
It was easy to change, and I retested with python 2 and 3 with success, so I committed this as you suggested.